internal/coreinternal/attraction/coercer.go

```diff
@@ -0,0 +1,69 @@
+package attraction
+
+import (
+	"regexp"
+	"strconv"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+var num = regexp.MustCompile(`^(\d+)(?:\.\d+)?$`)
+var dub = regexp.MustCompile(`^(\d+(?:\.\d+)?)$`)
+
+func coerceValue(to string, v pdata.AttributeValue) {
+	switch to {
+	case "string":
+		switch v.Type().String() {
+		case "STRING":
+		default:
+			v.SetStringVal(v.AsString())
+		}
+	case "int":
+		switch v.Type().String() {
+		case "INT":
+		case "DOUBLE":
+			v.SetIntVal(int64(v.DoubleVal()))
+		case "BOOL":
+			if v.BoolVal() {
+				v.SetIntVal(1)
+			} else {
+				v.SetIntVal(0)
+			}
+		case "STRING":
+			s := v.StringVal()
+			n := num.FindStringSubmatch(s)
```

Comment:
Why not just `strconv.ParseInt`? Is such approach more efficient?

Reply:
No good rationale for the existing code. I'm not very familiar with go and its idioms, so I'll adopt your suggestions

Reply:
Just out of curiosity, did run a quick benchmark:
```go
func BenchmarkTestRegexAtoi(b *testing.B) {
	var num = regexp.MustCompile(`^(\d+)(?:\.\d+)?$`)
	for n := 0; n < b.N; n++ {
		s := "1234"
		n := num.FindStringSubmatch(s)
		if n != nil {
			strconv.Atoi(n[1])
		}
	}
}

func BenchmarkTestParseInt(b *testing.B) {
	for n := 0; n < b.N; n++ {
		s := "1234"
		strconv.ParseInt(s, 10, 64)
	}
}
```

It seems the latter is much more efficient:
```
BenchmarkTestRegexAtoi
BenchmarkTestRegexAtoi-16    	 4305942	       257.4 ns/op
BenchmarkTestParseInt
BenchmarkTestParseInt-16     	85319960	        15.41 ns/op
```